syntax = "proto3";

package ssn.type;

option go_package = "type";

import "ssn/type/geometry.proto";

message Confidence {
  enum Level {
    UNKNOWN = 0;
    VERY_LOW = 1;
    LOW = 2;
    MID = 3;
    HIGH = 4;
    VERY_HIGH = 5;
  }
  // A bucketized representation of confidence, which is intended to give clients
  // highly stable results across model upgrades.
  Level level = 1;
  // The confidence value
  float value = 2;
}

message Candidate {
  // Normalized value 01-01-2019
  string value = 1;
  // The text as found on the document "1. Jan"
  // useful for overlays
  string text = 2;
  // Confidence
  Confidence confidence = 3;
  // The bounding box for the block.
  // The vertices are in the order of top-left, top-right, bottom-right,
  // bottom-left. When a rotation of the bounding box is detected the rotation
  // is represented as around the top-left corner as defined when the text is
  // read in the 'natural' orientation.
  // For example:
  //
  // * when the text is horizontal it might look like:
  //
  //         0----1
  //         |    |
  //         3----2
  //
  // * when it's rotated 180 degrees around the top-left corner it becomes:
  //
  //         2----3
  //         |    |
  //         1----0
  //
  //   and the vertex order will still be (0, 1, 2, 3).
  ssn.type.BoundingPoly bounding_box = 4;
}
